java-checks-test-sources/src/main/java/checks/security/EncryptionAlgorithmCheck.java

```diff
@@ -13,41 +13,41 @@ abstract class EncryptionAlgorithmCheck {
   public void foo(java.util.Properties props) {
     /*
     should complain:
-    - everytime ECB mode is used whatever the encryption algorithm
+    - everytime ECB mode is used
        - By default without specifying operation mode ECB is chosen
-    - when CBC mode is used with PKCS5Padding or PKCS7Padding
+    - when CBC mode is used with padding different from "NoPadding"
     - when RSA is used without OAEPWithSHA-1AndMGF1Padding or OAEPWITHSHA-256ANDMGF1PADDING padding scheme
     */
 
     try {
       // First case
-      Cipher.getInstance("AES"); // Noncompliant [[sc=26;ec=31]] {{Use secure mode and padding scheme.}}
+      Cipher.getInstance("AES"); // Noncompliant [[sc=26;ec=31]] {{Use a secure padding scheme.}}
 
-      Cipher.getInstance("AES/ECB/NoPadding"); // Noncompliant
-      Cipher.getInstance("AES" + "/ECB/NoPadding"); // Noncompliant
-      Cipher.getInstance("AES/ECB/NoPadding", getProvider()); // Noncompliant
-      Cipher.getInstance("AES/ECB/NoPadding", "someProvider"); // Noncompliant
+      Cipher.getInstance("AES/ECB/NoPadding"); // Noncompliant {{Use a secure cipher mode.}}
```

Comment:
It's not mandatory to check the message every time. You can test put just `// Noncompliant`